src/context/ChangeDataContext.tsx

```diff
@@ -0,0 +1,57 @@
+import { createContext, ReactNode, useEffect, useState } from 'react'
+import { information } from '../data/data'
+
+interface changeDataContextInterface {
+  movieData: Array<any>
+  movieID: number
+  deleteMovie?: (param: number) => void
+  newMovie: object
+  addNewMovie?: (param: any) => void
+}
+
+const defaultState = {
+  movieData: information,
+  movieID: 0,
+  newMovie: {},
+}
+
+
+
+export const ChangeDataContext = createContext<changeDataContextInterface>(defaultState)
+
+type Props = {
+  children: ReactNode
+}
+
+export const ChangeDataContextProvider = ({ children }: Props) => {
+
+
+  const [movieData, setMovieDescription] = useState(defaultState.movieData)
+  const [movieID, setMovieID] = useState(defaultState.movieID)
+  const [newMovie, setNewMovie] = useState(defaultState.newMovie)
+
```

Comment:
I think you need to handle this in redux itself. You can have a store and actions would update that store state